src/subdomains/core/history/controllers/transaction.controller.ts

```diff
@@ -113,57 +142,68 @@ export class TransactionController {
     });
   }
 
-  @Get('CoinTracking')
-  @ApiOkResponse({ type: CoinTrackingCsvHistoryDto, isArray: true })
+  @Get('target')
+  @ApiBearerAuth()
+  @UseGuards(AuthGuard(), new RoleGuard(UserRole.USER))
   @ApiExcludeEndpoint()
-  async getCsvCT(
-    @Query() query: HistoryQueryUser,
-    @Res({ passthrough: true }) res: Response,
-  ): Promise<CoinTrackingCsvHistoryDto[] | StreamableFile> {
-    if (!query.format) query.format = ExportFormat.CSV;
-    return this.getHistoryData(query, ExportType.COIN_TRACKING, res);
+  async getTransactionTargets(@GetJwt() jwt: JwtPayload): Promise<TransactionTarget[]> {
+    const buys = await this.buyService.getUserDataBuys(jwt.id);
+
+    return buys.map((b) => ({
+      id: b.id,
+      address: b.user.address,
+      asset: AssetDtoMapper.toDto(b.asset),
+      bankUsage: b.bankUsage,
+    }));
   }
 
-  @Get('ChainReport')
-  @ApiOkResponse({ status: 200, type: ChainReportCsvHistoryDto, isArray: true })
+  @Put(':id/target')
+  @ApiBearerAuth()
+  @UseGuards(AuthGuard(), new RoleGuard(UserRole.USER))
   @ApiExcludeEndpoint()
-  async getCsvChainReport(
-    @Query() query: HistoryQueryUser,
-    @Res({ passthrough: true }) res: Response,
-  ): Promise<ChainReportCsvHistoryDto[] | StreamableFile> {
-    if (!query.format) query.format = ExportFormat.CSV;
-    return this.getHistoryData(query, ExportType.CHAIN_REPORT, res);
+  async setTransactionTarget(
+    @GetJwt() jwt: JwtPayload,
+    @Param('id') id: string,
+    @Query('buyId') buyId: string,
+  ): Promise<void> {
+    const transaction = await this.transactionService.getTransaction(+id, { bankTx: true });
+    if (!transaction.bankTx) throw new NotFoundException('Transaction not found');
+    if (!BankTxTypeUnassigned(transaction.bankTx.type)) throw new ConflictException('Transaction already assigned');
+
+    const user = await this.userService.getUser(jwt.id, { userData: true });
+    const bankDatas = await this.bankDataService.getBankDatasForUser(user.userData.id);
+    if (!bankDatas.some((b) => b.iban === transaction.bankTx.iban))
+      throw new ForbiddenException('You can only assign your own transaction');
+
+    await this.bankTxService.update(transaction.bankTx.id, { type: BankTxType.BUY_CRYPTO, buyId: +buyId });
```

Comment:
buyId should be checked before, or? So if the buyId is existing and if the userData is the same with jwt userData